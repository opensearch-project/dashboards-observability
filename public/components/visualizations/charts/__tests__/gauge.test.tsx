/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import {
  GAUGE_TEST_VISUALIZATIONS_DATA,
  LAYOUT_CONFIG,
} from '../../../../../test/event_analytics_constants';
import { Gauge } from '../financial/gauge/gauge';

describe('Gauge component', () => {
  configure({ adapter: new Adapter() });

  it('Renders gauge component', async () => {
    const wrapper = mount(
      <Gauge visualizations={GAUGE_TEST_VISUALIZATIONS_DATA} layout={LAYOUT_CONFIG} config={true} />
    );

    wrapper.update();

    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
