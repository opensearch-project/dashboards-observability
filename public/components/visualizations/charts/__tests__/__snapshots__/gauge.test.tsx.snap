// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gauge component Renders gauge component 1`] = `
<div
  className="euiText euiText--extraSmall visWorkspaceNoData"
  data-test-subj="vizWorkspace__noData"
>
  <div
    className="euiTextAlign euiTextAlign--center"
  >
    <div
      className="euiTextColor euiTextColor--subdued"
    >
      <svg
        aria-hidden={true}
        className="euiIcon euiIcon--xxLarge euiIcon--subdued euiIcon-isLoading"
        focusable="false"
        height={16}
        role="img"
        style={null}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      />
      <div
        className="euiSpacer euiSpacer--l"
      />
      <p>
        <span>
          No data found
        </span>
      </p>
    </div>
  </div>
</div>
`;
